### PR TITLE
fixed flyout cannot be light dismissed by clicking outside of its content

### PR DESCRIPTION
**Pull request by XiaotianNetlift**
GitHub Issue (If applicable): #2677

## PR Type
What kind of change does this PR introduce?
- Bugfix

## What is the current behavior?
The light-dismiss currently works, but only if you click outside of the panel itself. However, when using "Placement=Full", this panel occupies the entire screen, thus making it impossible to click outside of it.

## What is the new behavior?
 You can now click on the area not occupied by the "Flyout.Content" to dismiss it.

## PR Checklist
Please check if your PR fulfills the following requirements:
- [x] Tested code with current [supported SDKs](https://github.com/unoplatform/uno/blob/master/README.md#uno-features)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Updated the [Release Notes](https://github.com/unoplatform/uno/tree/master/doc/ReleaseNotes)
- [x] Associated with an issue (GitHub or internal)

## Other information
Internal Issue (If applicable):
https://nventive.visualstudio.com/Umbrella/_workitems/edit/167567

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Controls/Flyout/FlyoutPresenter.cs [hunk 1]
```diff
@@ -10,8 +14,19 @@ public FlyoutPresenter()
...
-			// All pointer-related should be "eaten" to prevent closing
-			// the flyout when a tap is done in its content
+			if (this.GetTemplateRoot() is FrameworkElement root && this.Parent is FlyoutBasePopupPanel panel)
+			{
+				// allow flyout to be closed by clicking outside its content
+				var pointer = args.GetCurrentPoint(this);
+				var rootCoords = this.TransformToVisual(root).TransformPoint(pointer.Position);
+
+				if (0 > rootCoords.X || rootCoords.X > root.ActualWidth ||
+					0 > rootCoords.Y || rootCoords.Y > root.ActualHeight)
```
Comment (carldebilly) [inline src/Uno.UI/UI/Xaml/Controls/Flyout/FlyoutPresenter.cs]: Ask @dr1rrb, we probably have helpers for that.

**Comment by jeromelaban** [comment]
Thanks ! Could you augment this set of tests ?

https://github.com/unoplatform/uno/blob/f064db49c507fda7533d1465112dabd884cf8fb0/src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/FlyoutTests/UnoSamples_Tests.Flyout.cs#L16

File: src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/FlyoutTests/UnoSamples_Tests.Flyout.cs [hunk 2]
```diff
@@ -134,5 +134,61 @@ public void FlyoutTest_Unloaded()
...
+			// initial state
+			_app.WaitForElement(testableFlyoutButtons.First().Value);
+			var initialScreenshot = TakeScreenshot($"{majorStepIndex++} Initial State", ignoreInSnapshotCompare: true);
+
+			var dismissArea = GetDismissAreaCenter();
+			foreach (var button in testableFlyoutButtons)
+			{
+				// show flyout
+				button.Value.Tap();
+				var flyoutOpenedScreenshot = TakeScreenshot($"{majorStepIndex++} {button.Key} 0 Opened", ignoreInSnapshotCompare: true);
```
Comment (jeromelaban) [inline src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/FlyoutTests/UnoSamples_Tests.Flyout.cs]: Would be useful to add the key there:

```suggestion
				var flyoutOpenedScreenshot = TakeScreenshot($"{majorStepIndex++} {button.Key} {button.Key} 0 Opened", ignoreInSnapshotCompare: true);
```

**Comment by jeromelaban** [comment]
Make sure to validate the screenshots for changes over popups/flyouts.

File: src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/FlyoutTests/UnoSamples_Tests.Flyout.cs (hunk 2 above)
Comment (XiaotianNetlift) [reply src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/FlyoutTests/UnoSamples_Tests.Flyout.cs]: It is already there. You just duplicated it.

Comment (XiaotianNetlift) [inline src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/FlyoutTests/UnoSamples_Tests.Flyout.cs]: `majorStepIndex` should only be incremented at end of loop.

**Comment by XiaotianNetlift** [comment]
todo: 
- [ ] wasm: fix centered full flyout not dismissed or disable it for wasm
- [ ] doird: find another workaround for the status bar problem

**Comment by dr1rrb** [comment]
Humm, I'm pretty sure I've open an issue for this, but I'm not able to find it.

This is not the right way to fix the bug, we should implement the [`Popup.IsLightDismissEnabled` property](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.primitives.popup.islightdismissenabled)

You approach requires to stretch the `FlyoutPresenter` (and the parent `Popup`), which is known to be problematic. So we can go forward with this fix, but it's a temporary fix ... and we would assume that we will get some other issues/regression.

File: src/Uno.UI/UI/Xaml/Controls/Flyout/FlyoutPresenter.cs [hunk 3]
```diff
@@ -10,8 +14,19 @@ public FlyoutPresenter()
 
 		protected override void OnPointerPressed(PointerRoutedEventArgs args)
 		{
-			// All pointer-related should be "eaten" to prevent closing
-			// the flyout when a tap is done in its content
+			if (this.GetTemplateRoot() is FrameworkElement root && this.Parent is FlyoutBasePopupPanel panel)
+			{
+				// allow flyout to be closed by clicking outside its content
+				var pointer = args.GetCurrentPoint(this);
```
[the remark this answers is not in the document]
Comment (dr1rrb) [reply src/Uno.UI/UI/Xaml/Controls/Flyout/FlyoutPresenter.cs]: This is useless on event handlers.

File: src/Uno.UI/UI/Xaml/Controls/Flyout/FlyoutPresenter.cs (hunk 1 above)
Comment (dr1rrb) [reply src/Uno.UI/UI/Xaml/Controls/Flyout/FlyoutPresenter.cs]: Not sure for this one ... maybe 🤷‍♂️

File: src/Uno.UI/UI/Xaml/Controls/Flyout/FlyoutPresenter.cs [hunk 4]
```diff
@@ -10,8 +14,19 @@ public FlyoutPresenter()
...
+				var rootCoords = this.TransformToVisual(root).TransformPoint(pointer.Position);
```
Comment (dr1rrb) [inline src/Uno.UI/UI/Xaml/Controls/Flyout/FlyoutPresenter.cs]:
```suggestion
				var rootCoords = args.GetCurrentPoint(root).Position;
```

File: src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/FlyoutTests/UnoSamples_Tests.Flyout.cs [hunk 5]
```diff
@@ -121,18 +121,63 @@ public void FlyoutTest_Target()
 		[AutoRetry]
 		public void FlyoutTest_Unloaded()
 		{
-			Run("UITests.Shared.Windows_UI_Xaml_Controls.Flyout.Flyout_Unloaded");
+			_app.Repl();
```
Comment (Xiaoy312) [inline src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/FlyoutTests/UnoSamples_Tests.Flyout.cs]: debug code...

**Pull request merged**